pkgs/by-name/gj/gjs/package.nix

```
@@ -95,22 +95,26 @@ stdenv.mkDerivation (finalAttrs: {
 
   nativeCheckInputs = [
     xvfb-run
-  ]
-  ++ testDeps;
+  ];
+
+  checkInputs = testDeps;
 
   propagatedBuildInputs = [
     glib
   ];
 
   mesonFlags = [
     "-Dinstalled_test_prefix=${placeholder "installedTests"}"
+    (lib.mesonBool "skip_gtk_tests" (!finalAttrs.finalPackage.doCheck))
```

Shouldn't
```suggestion
    (lib.mesonBool "skip_gtk_tests" (!finalAttrs.doCheck))
```
also work with `overrideAttrs`?

---

unfortunately `stdenv.mkDerivation` itself modifies `doCheck` in a way that's visible on the `finalPackage` but not the `finalAttrs`. particularly [this](https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/make-derivation.nix#L388), [this](https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/make-derivation.nix#L474) and [this](https://github.com/NixOS/nixpkgs/blob/master/pkgs/stdenv/generic/make-derivation.nix#L616) line.

those 3 lines in stdenv.mkDerivation are relevant for cross compilation: you can see the hash of `pkgsCross.aarch64-multiplatform.gjs` change between this line being `finalAttrs.finalPackage.doCheck` (evaluates to `false`) v.s. `finalAttrs.doCheck` (evaluates to _true_ -- despite that `mkDerivation` behaves as if it were false, by stripping out `checkInputs` etc).

so using `finalAttrs.finalPackage.doCheck` here is actually significant & intentional.